### ISSUE TITLE
Default Code Question

### ISSUE THREAD

**Issue by wrldtvlr** [josephdadams/TallyArbiter-M5AtomMatrixListener#7]
In the Matrix ino file, there is a block of initialization code as follows:

```
  preferences.begin("tally-arbiter", false);
  if (preferences.getString("deviceid").length() > 0) {
    DeviceId = preferences.getString("deviceid");
    //DeviceId = "unassigned";
  }
  if (preferences.getString("devicename").length() > 0) {
    DeviceName = preferences.getString("devicename");
    //DeviceName = "unassigned";
  }
  preferences.end();
```

In the repo version, the first lines are commented out such that the M5 is unassigned. This requires that the M5 be configured by pushing the button on every powerup. Am I correct in understanding that if I swap the commented lines as shown above, that the device will inherit whatever assignment it had previously?

**Comment by wrldtvlr** [comment]
I've done some preliminary testing and this code change does seem to work better. Must do further testing to see if there are any unwanted side effects. I'm also comparing this code to the code for the M5_StackC listener. On first glance, it seems that code might be a little more advanced. It might be useful to take that code and swap in the Matrix display function.

```
  preferences.begin("tally-arbiter", false);
  if (preferences.getString("deviceid").length() > 0) {
    DeviceId = preferences.getString("deviceid");}
    else {DeviceId = "unassigned";}
  
  if (preferences.getString("devicename").length() > 0) {
    DeviceName = preferences.getString("devicename");}
    else {DeviceName = "unassigned";}
  
  preferences.end(); 
```

**Comment by jerriley** [comment]
Yep I'm not sure why this bit of code was changed as compared with that of the M5StickC. I replaced it with that of the M5StickC and now it remembers its Device following a reboot.

**Comment by jerriley** [comment]
I had some issues today with re-installing tallyarbiter and then my m5-atom not being recognized. I started from scratch and will try the code posted by wrldtvlr above. I suspect it may be related. It's working again now.

**Comment by wrldtvlr** [comment]
@jerriley We've encountered the same problem. The code change I suggested up thread works as long as the server has recognized the listener previously. If the server has been restarted with no previous memory, it will not recognize listeners that present a unique ID the server hasn't issued. There are probably a couple of ways to address this. Just trying to figure out the best approach.

**Comment by themusicnerd** [comment]
Sorry I half/mostly abandoned the project for a while.
Great pickup!
This was something I changed when testing a whole heap of things in a semi-chaotic manner.
I'm amending it now.

**Comment by wrldtvlr** [comment]
Excellent news - thank you for continuing to work on this!
